### ISSUE TITLE
1.5.0-17: Now ender affinity reduces mana cost on ALL spells (cmon man)

### ISSUE THREAD

**Issue by ProfHugo** [Growlith1223/ArsMagica2#175]
Firstly, #60 , I stated how affinity had no effect on mana cost... and I gave an example of that, with ender affinity.

But, somehow
```
float enderDepth = (float)AffinityData.For((EntityLivingBase)caster).getAffinityDepth(Affinity.ENDER);
float reduction = 1 - (0.75f * enderDepth);
cost = cost * reduction;
```
For context, according to the "Mithion Bible", "Components of the same affinity as you will cost less mana and depending on your affinity depth, may not even cost reagents. Components of opposing affinity, however, will cost more mana and again, depending on your affinity depth, may cost additional reagents." 

This means that the way mana cost is calculated, is the deeper your affinity is in a certain element (say fire), the less mana u need to use in order to cast fire spells, while mana cost for spells with affinities that are "opposing" fire will increase according to the map in the occulus (water and life for fire).

In short, 2f5e136 is just plain ridiculous, as I don't meant that ender affinity should be chopping down manacost on ALL SPELLS, so highly mobile (tricky, slippery, and evil) ender mages could also enjoy the luxury of using more than 4 times the mana than their competitors, while outclassing all other mages just because an ender mage could use 4 fireballs with the same mana as a firemage casting one. I'm sure there was something relating to affinity to mana cost calculation somewhere in the code, but at least, do something about these game-breaking endermage buffs and follow the bible...

Note: Mithion Bible = Arcane Compendium (http://am2.wikia.com/wiki/Affinities for the lazy ones)

**Comment by Growlith1223** [comment]
yea my bad, was tired when i was doing the fix and misread the post, working on a real fix for this now lol

**Comment by Growlith1223** [comment]
fixed!

**Comment by ProfHugo** [comment]
I guess maybe just having affinities reducing mana cost on their respective components might be good enough, but maybe implementing penalties for using opposing spells of your affinity might be for the better (like increased mana cost on water and life based spells if you're a firemage, just like AM2 1.7.10).
